src/main/config/config.c

```diff
@@ -555,6 +560,34 @@ static void validateAndFixConfig(void)
 #if defined(TARGET_VALIDATECONFIG)
     targetValidateConfiguration();
 #endif
+
+    for (unsigned i = 0; i < CONTROL_RATE_PROFILE_COUNT; i++) {
+        switch (currentControlRateProfile->rates_type) {
+        case RATES_TYPE_BETAFLIGHT:
+        default:
+            for (int axis = FD_ROLL; axis <= FD_YAW; axis++) {
+            currentControlRateProfile->rates[axis] = constrain(currentControlRateProfile->rates[axis], 0, BETAFLIGHT_MAX_SRATE);
```

Comment:
You need to use `controlRatesProfileMutable(i)` here instead of `currentControlRateProfile` or you'll do the checks multiple times on the same profile.

Reply:
Ops.